File: Retos/Reto #4 - PRIMO, FIBONACCI Y PAR [Media]/python/mariovelascodev.py
def what_is_it(number):
    #Variables
    result = "{number} ".format(number=number)
    fibonacci = [1, 1]
    count = 1

    try:
    
        #Comprobamos si el número es primo
        for index in range(2, number):
            if number % index == 0:
                result += "no es primo, "
                break
        else:
            result += "es primo, "

        #Comprobamos si es fibonacci
        #Sacamos la secuencia de Fibonacci del rango introducido por parametro y añadimos a la lista
        while count < number:
            next_number = fibonacci[count] + fibonacci[count - 1]
            fibonacci.append(next_number)

            #Si el último número es igual al introducido por parámetros paramos el bucle
            if next_number == number:
                break
            
            count+=1

        #Comprobamos si el último número es igual al introducido por parámetros
        if fibonacci[len(fibonacci)-1] == number:
            result += "es fibonacci"
        else:
            result += "no es fibonacci"

        #Comprobamos si el número es par
        if number % 2 == 0:
            result += " y es par"
        else:
            result += " y es impar"
        
        return result
    except TypeError:
        print("Debes introducir un número")

print(what_is_it(2))
print(what_is_it(7))
print(what_is_it(4))
print(what_is_it(0))
print(what_is_it(98))

File: Retos/Reto #4 - PRIMO, FIBONACCI Y PAR [Media]/javascript/mariovelascodev.js
"use strict"

function whatIsIt(number) {

    //Variables
    let result = "";
    let fibonacci = [1, 1];

    //Comprobar si el número introducido es par
    if (number % 2 === 0) {
        result += "par";
    }else {
        result += "impar";
    }

    //Comprobar si el número es primo
    if(number % 2 === 0 && number % 3 === 0 && number % 5 === 0 && number % 7 === 0 && number % 11 === 0) {
        result += ", no es primo";
    }else {
        result += ", es primo";
    }

    //Comprobar Fibonacci
    
    //Sacamos la secuencia de Fibonacci del rango introducido por parametro y añadimos al array
    for(let i=1 ; i < number; i++) {
        let nextNumber = fibonacci[i] + fibonacci[i-1];
        fibonacci.push(nextNumber);

        //Si el último número es igual al introducido por parámetros paramos el bucle
        if (nextNumber === number) {
            break;
        }
    }

    //Comprobamos si el último número es igual al introducido por parámetros
    if(fibonacci[fibonacci.length-1] === number) {
        result += ", es fibonacci";
    }else {
        result += ", no es fibonacci";
    }

    console.log(number+" es "+result);
}

whatIsIt(2);
whatIsIt(7);
whatIsIt(13);
whatIsIt(113);